gui/wxpython/gui_core/dialogs.py

```diff
@@ -1107,8 +1107,13 @@ def OnAddLayer(self, event):
 
     def OnRemoveLayer(self, event):
         """Remove layer from listbox"""
-        while self.gLayerBox.GetSelections():
-            sel = self.gLayerBox.GetSelections()[0]
+        # After removal of last selected item by .Delete,
+        # ListBox selects the last of remaining items in the list
+        # and thus adds a new item to GetSelections
+        # Items are removed in reverse order to maintain positional number
+        # of other selected items (ListBox is dynamic!)
+        selections = reversed(sorted(self.gLayerBox.GetSelections()))
```

Comment:
[`sorted()`](https://docs.python.org/3/library/functions.html#sorted) func has `reverse` parameter, `reversed()` function is no needed here.

Reply:
Also an option.